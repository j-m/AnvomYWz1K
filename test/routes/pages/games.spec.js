'use strict'

const request = require('supertest')

const app = require('../../../src/app/koa')
const connection = require('../../../src/database/connection')

beforeAll(async() => {
  jest.resetModules()
  process.env.DATABASE = ':memory:'
  await connection.open()
})

afterAll(async() => {
  await connection.close()
})

describe('routes get games', () => {
  test('should contain templated text', async() => {
    const response = await request(app.callback()).get('/games/')
    expect(response.text).toMatchSnapshot()
  })
})
