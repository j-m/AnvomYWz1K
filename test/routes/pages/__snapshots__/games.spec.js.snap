// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes get games should contain templated text 1`] = `
"<!DOCTYPE html>

<html lang=\\"en\\">
	<head>
		<meta charset=\\"utf-8\\">
		<title>Game Reviews - Games</title>
    <link href=\\"https://fonts.googleapis.com/css?family=Roboto&display=swap\\" rel=\\"stylesheet\\">
    
    <link href=\\"../css/global.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    <script src=\\"../js/global.js\\" type=\\"text/javascript\\"></script>
    
		<link href=\\"../css/loginRegistration.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    <script src=\\"../js/loginRegistration.js\\" type=\\"text/javascript\\"></script>
    
    <link href=\\"../css/games.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    <script src=\\"../js/games.js\\" type=\\"text/javascript\\"></script>
	</head>
  <body>
    <div id=\\"welcome\\">
      <p>Welcome, guest! <a href=\\"#\\" onclick=\\"show('login')\\">Login</a> or <a href=\\"#\\" onclick=\\"show('register')\\">Register</a> </p>
    </div>
    
    <div class=\\"popup\\" onclick=\\"hide('login', event)\\">
      <form id=\\"login\\" onsubmit=\\"login(event)\\">
        <div class=\\"input_header\\">
        	<label for=\\"username\\">Username</label>
          <p class=\\"error USERNAME_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"text\\" id=\\"username\\" name=\\"username\\" autocomplete=\\"current-username\\" required/>
        <p class=\\"error USERNAME_UNKNOWN\\">This username does not exist. </p>
        
        <div class=\\"input_header\\">
        	<label for=\\"password\\">Password</label>
          <p class=\\"error PASSWORD_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"password\\" id=\\"password\\" name=\\"password\\" autocomplete=\\"current-password\\" required/>
        <p class=\\"error PASSWORD_INCORRECT\\">Incorrect password. </p>
        
      	<input type=\\"submit\\" value=\\"Log in\\"/>
        <a href=\\"#forgot\\">Forgot password</a>
      </form>
    </div>
    
    <div class=\\"popup\\" onclick=\\"hide('register', event)\\">
      <form id=\\"register\\" onsubmit=\\"register(event)\\">
        <div class=\\"input_header\\">
        	<label for=\\"email\\">Email</label>
          <p class=\\"error EMAIL_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"email\\" id=\\"email\\" name=\\"email\\" autocomplete=\\"email\\" required/>
        
        <div class=\\"input_header\\">
        	<label for=\\"new-username\\">Username</label>
          <p class=\\"error USERNAME_MISSING\\">This field is required. </p>
          <p class=\\"error USERNAME_AVAILABLE\\">Username available. </p>
        </div>
        <input type=\\"text\\" id=\\"new-username\\" name=\\"new-username\\" autocomplete=\\"none\\" required/>
        <button type=\\"button\\" onclick=\\"checkUsernameAvailability()\\">&#x21bb;</button>
        <p class=\\"error USERNAME_IN_USE\\">This username is already in use. </p>
        <p class=\\"info USERNAME_BAD_REGEX\\">Usernames can only contain letters and numbers. <br/> They may contain, but not start or end with, hyphens(-). </p>
        
        <div class=\\"input_header\\">
        	<label for=\\"new-password\\">Password</label>
          <p class=\\"error PASSWORD_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"password\\" id=\\"new-password\\" name=\\"new-password\\" minlength=\\"10\\" autocomplete=\\"new-password\\" required/>
        <p class=\\"info PASSWORD_TOO_SHORT\\">Passwords must be at least 10 characters long. </p>
        
      	<input type=\\"submit\\" value=\\"Register\\">
      </form>
    </div>
    
    <div id=\\"games\\">
    </div>

    <div class=\\"popup\\" onclick=\\"hide('addGame', event)\\">
      <form id=\\"addGame\\" onsubmit=\\"addGame(event)\\">
        <h1>Create new game</h1>
        <hr/>
        <fieldset>
          <div class=\\"input_header\\">
          	<label for=\\"steamAppID\\">Steam App ID</label>
            <p class=\\"error GAME_STEAM_APP_ID_MISSING\\">This field is required</p>
          </div>
          <input type=\\"text\\" id=\\"steamAppID\\" name=\\"steamAppID\\" required/>
          
          <div class=\\"input_header\\">
          	<label for=\\"title\\">Title</label>
            <p class=\\"error GAME_TITLE_MISSING\\">This field is required</p>
          </div>
          <input type=\\"text\\" id=\\"title\\" name=\\"title\\" required/>
          <p class=\\"error GAME_TITLE_IN_USE\\">Title in use</p>
        
          <div class=\\"input_header\\">
          	<label for=\\"summary\\">Summary</label>
            <p class=\\"error GAME_SUMMARY_MISSING\\">This field is required</p>
          </div>
          <input type=\\"text\\" id=\\"summary\\" name=\\"summary\\" required/>
        </fieldset>
        
        <fieldset>
          <legend>Optional</legend>
          
          <div>
          	<label for=\\"developer\\">Developer</label>
            <input type=\\"text\\" id=\\"developer\\" name=\\"developer\\"/>
            
          	<label for=\\"publisher\\">Publisher</label>
            <input type=\\"text\\" id=\\"publisher\\" name=\\"publisher\\"/>
            
          	<label for=\\"description\\">Description</label>
            <input type=\\"text\\" id=\\"description\\" name=\\"description\\"/>
          </div>
          
          <div>
          	<label for=\\"store\\">Store</label>
            <input type=\\"text\\" id=\\"store\\" name=\\"store\\"/>

            <label for=\\"releaseDate\\">Release date</label>
            <input type=\\"date\\" id=\\"releaseDate\\" name=\\"releaseDate\\"/>
            
            <label for=\\"tags\\">Tags</label>
            <input type=\\"text\\" id=\\"tags\\" name=\\"tags\\"/>
          </div>

        </fieldset>
        
      	<input type=\\"submit\\" value=\\"Create\\">
      </form>
    </div>
    
  </body>

</html>
"
`;
